rdmo/core/pandoc.py

```
@@ -23,8 +23,9 @@ def get_pandoc_content(html, metadata, export_format, context):
     pandoc_args = get_pandoc_args(export_format, context)
 
     if metadata:
-        # create a temporary file for the metadata
+        # create a temporary file for the metadata and close it immediately
         (metadata_tmp_fd, metadata_tmp_file_name) = mkstemp(suffix='.json')
+        os.close(metadata_tmp_fd)
```

please just use context managers, that's what they are for and to prevent this forgetting of closing when opening a file

my LLM-assisted suggestion:
```py
from tempfile import NamedTemporaryFile
from pathlib import Path

def get_pandoc_content(html, metadata, export_format, context):
    pandoc_args = get_pandoc_args(export_format, context)

    metadata_tmp_path = None
    output_tmp_path = None

    try:
        # Handle metadata file
        if metadata:
            with NamedTemporaryFile('w+', suffix='.json', delete=False) as metadata_file:
                json.dump(metadata, metadata_file)
                metadata_file.flush()
                metadata_tmp_path = Path(metadata_file.name)
                log.info('Saved metadata file %s %s', metadata_tmp_path, str(metadata))
                pandoc_args.append(f'--metadata-file={metadata_tmp_path}')

        # Handle output file
        with NamedTemporaryFile('wb+', suffix=f'.{export_format}', delete=False) as output_file:
            output_tmp_path = Path(output_file.name)

        log.info('Export %s document using args %s.', export_format, pandoc_args)

        html = re.sub(
            r'(<img.+src=["\'])' + settings.STATIC_URL + r'([\w\-\@?^=%&/~\+#]+)',
            r'\g<1>' + str(Path(settings.STATIC_ROOT)) + r'/\g<2>',
            html
        )

        pypandoc.convert_text(html, export_format, format='html',
                              outputfile=str(output_tmp_path), extra_args=pandoc_args)

        pandoc_content = output_tmp_path.read_bytes()
        return pandoc_content

    finally:
        # Cleanup temp files
        if metadata_tmp_path and metadata_tmp_path.exists():
            metadata_tmp_path.unlink()
        if output_tmp_path and output_tmp_path.exists():
            output_tmp_path.unlink()
```

---

should we also catch a `RunTimeError` when `pypandoc.convert_text` fails?
```py
try:
    ...
    pypandoc.convert_text(html, export_format, format='html', outputfile=tmp_file_name, extra_args=pandoc_args)
    ...
except RuntimeError as e:
    log.error('Failed to convert text using pandoc: %s', e)
    ...
    # raise of return ?
```

---

No it should just raise the usual error.

---

but what about 500 errors??